docs/_posts/2020-02-07-boltwash.md

```diff
@@ -0,0 +1,128 @@
+```
+wash bolt > tree -L 2 my-linux-nodes
+my-linux-nodes
+├── node1
+│   └── fs
+└── node2
+    └── fs
+wash bolt > cd my-linux-nodes/node1/fs
+wash bolt/my-linux-nodes/node1/fs > ls
+bin/
+boot/
+dev/
+etc/
+home/
+...
+```
+
+Great, I can see the files on that system, and I didn't have to do any additional setup. I can go look at config or watch logs
+```
+wash bolt/my-linux-nodes/node1/fs > cat etc/hosts
+127.0.0.1 localhost
+...
+wash bolt/my-linux-nodes/node1/fs > tail -f var/log/syslog
+Feb  7 18:39:51 pe-agent01 systemd[18116]: Startup finished in 23ms.
+Feb  7 18:39:51 pe-agent01 systemd[1]: Started User Manager for UID 1000.
+...
+```
+
+We can also explore Windows targets using WinRM
+```
+wash bolt/my-windows-nodes/node5/fs > cat Users
+Administrator/
+Public/
+```
+
+## Debugging with Bolt + Wash
+
+Wash can be helpful debugging all sorts of failures. As an example, suppose you have a cluster of machines (in the `webservers` group) that are having problems
+```
+wash bolt/webservers > ls
+server1/
+server2/
+server3/
+```
+
+You've noticed an issue in one of their logs, and want to see if it's present everywhere
+```
+wash bolt/webservers > grep OutOfMemory */fs/var/log/syslog
+server1/fs/var/log/syslog:Jan 28 12:09:34 server1 my-server[435]: ...java.lang.OutOfMemoryError...
+server2/fs/var/log/syslog:Jan 28 12:09:43 server2 my-server[435]: ...java.lang.OutOfMemoryError...
+server3/fs/var/log/syslog:Jan 28 12:09:01 server3 my-server[435]: ...java.lang.OutOfMemoryError...
+```
+
+Seeing that they're all experiencing these issues, let's see where the server process is still running
+```
+wash bolt/webservers > wps * | grep bin/java
+server1     9027   12:10.02   /usr/bin/java -Xmx2048m -cp /opt/my-server/server.jar ...
+server2    21204   12:13.02   /usr/bin/java -Xmx2048m -cp /opt/my-server/server.jar ...
+```
+
+Looks like they've restarted recently, and the server's not running on `server3`. Let's reconfigure the servers to allow more memory (using Bolt or by directly modifying config/service files) and restart them
+```
+wash bolt/webservers > bolt task run service -t webservers action=restart name=my-server
+Started on server1...
+Started on server2...
+Started on server3...
+Finished on server1:
+  {
+    "status": "MainPID=751,LoadState=loaded,ActiveState=active"
+  }
+Finished on server2:
+  {
+    "status": "MainPID=2927,LoadState=loaded,ActiveState=active"
+  }
+Finished on server3:
+  {
+    "status": "MainPID=21936,LoadState=loaded,ActiveState=active"
+  }
+Successful on 3 targets: server1,server2,server3
+Ran on 3 targets in 1.91 sec
+```
+
+You can imagine other scenarios as well. Suppose you ran a Bolt plan across several targets and want to get some files back from them. You can use Wash to directly copy them with something like
+```
+wash bolt > for f in group/*/fs/path/to/file; do cp -v "$f" ~/dest/"${f//\//_}"; done
```

Comment:
I'd maybe add a note here saying that the sed stuff is just to remove `fs` entry from the paths.